### DISCUSSION TITLE
maxLFQ.do function

### DISCUSSION THREAD

**Discussion by MarjorieLeduc** [tvpham/iq#21] [Q&A] [answered]
Hi,
I am studying iq maxLFQ.do function and I have difficulties to understand the ligne 334 of iq.R file:
        res <- lsfit(rbind(cbind(2*AtA , rep(1, N)),
                           c(rep(1, N), 0)),
                     c(2*Atb, mean(data.matrix(X), na.rm = TRUE) * N),
                     intercept = FALSE)

I wondered why is it needed to multiply by 2 AtA and Atb matrix before the linear regression ?
Thank you
Marjorie

**Comment by tvpham** [comment, marked as answer]
Hi Marjorie,

It is eq. 2 in the manuscript https://academic.oup.com/bioinformatics/article/36/8/2611/5697917.

There is a bit of mathematics to go from the minimization problem to eq. 2. You can find it in many textbook. I can write more about it also (but not very soon).

Thang

**Reply by MarjorieLeduc** [reply]
Hi,
Thank you very much for your answer.
Actualy, I compared iq maxlfq with diann_maxlfq on the same set of data and looked the log2(intensity) results for the same sample.
here is the plot :
![Capture d’écran 2024-09-13 133650](https://github.com/user-attachments/assets/02499783-3206-4944-95be-fc76e26a0429)
pink line is x=y, white line is x*2-13.
I wondered the difference between the 2 could come from this 2 factor and the c constant in eq. 2 ?
I am sorry I am not a mathematician, maybe it's a sensless question.
Thank you,
Marjorie

**Comment by tvpham** [comment]
Hi Marjorie,

I saw your question in the issue list. But I will need some time to look into that. Perhaps when DIA-NN version 1.9.2 is out.

The constant c can be an factor. It should at least affect the offset (like -13 in your analysis). Nevertheless, it is relative quantification. So it is perhaps more important to check the quantification across samples for individual proteins.

Thang

**Reply by MarjorieLeduc** [reply]
You are all right ratios between 2 samples are nearly the same with iq and diann packages.
Thank you very much.
Marjorie
